src/GuiBase/Utils/KeyMappingManager.cpp

```diff
@@ -217,6 +298,118 @@ void KeyMappingManager::loadConfiguration( const char* filename ) {
     }
 }
 
+bool KeyMappingManager::saveConfiguration( const char* filename ) {
+    QString flnm;
+    if ( filename == nullptr ) { flnm = m_file->fileName(); }
+    else
+    { flnm = filename; }
+    QFile saveTo( flnm );
+    saveTo.open( QIODevice::WriteOnly );
+    QXmlStreamWriter stream( &saveTo );
+    stream.setAutoFormatting( true );
+    stream.setAutoFormattingIndent( 4 );
+    stream.setCodec( "ISO 8859-1" );
+    stream.writeStartDocument();
+    stream.writeComment( "\tRadium KeyMappingManager configuration file\t" );
+    stream.writeComment(
+        "\n<keymap context=\"thecontext\" action=\"theAction\" buttons=\"QButton\" "
+        "modifier=\"QModifier\" key=\"QKey\" wheel=\"boolean\"/>\n" );
+    QDomNode root = m_domDocument.documentElement();
+    while ( not root.isNull() )
```

Comment:
`not` seems not recognize by visual studio (at least on appveyor) maybe use `!` instead